addon/components/ember-thead/component.js

```diff
@@ -225,17 +226,19 @@ export default Component.extend({
     this.columnTree.set('enableReorder', this.get('enableReorder'));
   },
 
+  _onColumnsChange() {
+    this._updateColumnTree();
+    scheduleOnce('actions', this, this.fillupHandler);
```

Comment:
Without the `schedule`, modification via mutation will not pass tests because when `fillupHandler` calls `ensureWidthConstraint`, the root's subcolumn nodes have not been updated (i.e., it doesn't yet see that a column was removed)

Reply:
@bantic something in the logic of `_updateColumnTree` is scheduling, or causing a schedule, into actions then? Did you hunt down what this is, and can you point me at it? I looked at [`_updateColumnTree`](https://github.com/Addepar/ember-table/blob/a67a2b1c4fa2fc56b0cf0ef3b749a022c361b62e/addon/components/ember-thead/component.js#L217) and don't see anything scheduling.

Reply:
I thought this was just observer related timing problems.  Am I wrong @mixonic?

Reply:
I haven't looked into the downstream source of the problem. The code uses what seems like an intermediate "caching" layer and I am not clear on how it works — the problem that this `scheduleOnce` fixes is that the `ensureWidthConstraint` that is called (by `fillupHandler`) looks at a different property to get the number of columns: the `column-tree`'s `root.subcolumnNodes`. That is the property that isn't in sync with `columns` until/unless we use the schedule.